templates/default/upstart.conf.erb

```diff
@@ -0,0 +1,52 @@
+# <%= @name %> generated by poise-service for <%= @new_resource.to_s %>
+
+description "<%= @name %>"
+
+start on runlevel [2345]
+stop on runlevel [!2345]
+
+<% limits = @new_resource.enclosing_provider.new_resource.limits %>
+<%= "limit nofile #{limits['open_files']} #{limits['open_files']}" %>
```

Comment:
Unlike the open file limit which is per-process, the process (i.e. thread) limit is global for the entire node, not sure it would be honored here.